src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -496,9 +507,9 @@ CreateRelationIsAKnownShardFilter(int pgClassVarno)
 	funcExpr->location = -1;
 	funcExpr->args = list_make1(oidVar);
 
-	BoolExpr *notExpr = makeNode(BoolExpr);
-	notExpr->boolop = NOT_EXPR;
-	notExpr->args = list_make1(funcExpr);
+	BooleanTest *notExpr = makeNode(BooleanTest);
+	notExpr->booltesttype = IS_NOT_TRUE;
+	notExpr->arg = (Expr *) funcExpr;
```

Comment:
FYI: This change is needed because the previous version `NOT relation_is_a_known_shard(oid)` would also filter out the row if oid was NULL, which was the case for some LEFT JOIN queries. This changes it to `relation_is_a_known_shard(oid) IS NOT TRUE`, which allows NULLs.

Reply:
To be clear, previously this worked fine because `securityQuals` are evaluated on the rows in the table before the join takes place, while regular `quals` are applied on the rows resulting from the join.